### ISSUE TITLE
[BUG][Python] Syntax Error when importing generated library

### ISSUE THREAD

**Issue by WilliamHYZhang** [OpenAPITools/openapi-generator#18030]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?

##### Description

Similar error as #17447, tested on both latest stable version 7.3.0 and latest master branch commit.

##### openapi-generator version

7.3.0

##### OpenAPI declaration file content or url

https://github.com/Endgrate/OpenAPI

##### Steps to reproduce

Copy OpenAPI spec

Run
```
openapi-generator-cli generate -g python -o test --additional-properties=projectName=endgrate,packageName=endgrate --openapi-normalizer SET_TAGS_FOR_ALL_OPERATIONS=endgrate -i OpenAPI/openapi.yaml
```

Install

```
pip install -e test
```

Import in Python
```
import endgrate
```

Gives:
```
line 26
    from endgrate.models.dict[str,_configuration_callback_schema_passthrough_schemas_value] import Dict[str, ConfigurationCallbackSchemaPassthroughSchemasValue]
                             ^
SyntaxError: invalid syntax
```

##### Related issues/PRs

#17447, seems like #17659 did not handle all edge cases? Does this have to do with beta support for OpenAPI 3.1?

**Comment by WilliamHYZhang** [comment]
Yeah, looking again I don't think it has to do with any specific 3.1 syntax...

**Comment by booxter** [comment]
I am able to reproduce this same failure with the following simple schema: https://github.com/booxter/letsrolld/commit/54ee9ae806f3b736bb6e28cc455f4c433660fe33
